unitary/examples/quantum_chinese_chess/move_test.py

```diff
@@ -261,3 +261,121 @@ def test_jump_basic():
+
+def test_split_jump_quantum_source():
+    # Source is in quantum state.
+    board = set_board(["a1"])
+    world = board.board
+    alpha.PhasedSplit()(world["a1"], world["a2"], world["a3"])
+    SplitJump()(world["a3"], world["a4"], world["a5"])
+    assert_sample_distribution(
+        board,
+        {
+            locations_to_bitboard(["a2"]): 0.5,
+            locations_to_bitboard(["a4"]): 0.25,
+            locations_to_bitboard(["a5"]): 0.25,
+        },
+    )
+    assert world["a4"].is_entangled == True
+    assert world["a5"].is_entangled == True
+
+
+def test_merge_jump_perfect_merge():
+    # Two quantum pieces split from one source could be merge back to one.
+    board = set_board(["a1"])
+    world = board.board
+    SplitJump()(world["a1"], world["a2"], world["a3"])
+    MergeJump()(world["a2"], world["a3"], world["a1"])
+    assert_samples_in(board, {locations_to_bitboard(["a1"]): 1.0})
+
+
+def test_merge_jump_imperfect_merge_scenario_1():
+    # Imperfect merge scenario 1
+    board = set_board(["a1"])
+    world = board.board
+    SplitJump()(world["a1"], world["a2"], world["a3"])
+    SplitJump()(world["a3"], world["a4"], world["a5"])
+    # a2 has prob. 0.5 to be occupied, while a4 has prob. 0.25 to be occupied
+    MergeJump()(world["a2"], world["a4"], world["a6"])
+    # Accoding to matrix calculations, the ending coefficient of
+    # a5 to be occupied: -1/2;
+    # a6 to be occupied: 1/2 + i/2/sqrt(2)
+    # a4 to be occupied: -i/2 -1/2/sqrt(2)
+    # a2 to be occupied: 0
+    assert_sample_distribution(
+        board,
+        {
+            locations_to_bitboard(["a5"]): 1.0 / 4,
+            locations_to_bitboard(["a6"]): 3.0 / 8,
+            locations_to_bitboard(["a4"]): 3.0 / 8,
+        },
+    )
+
+
+def test_merge_jump_imperfect_merge_scenario_2():
+    # Imperfect merge scenario 2
+    # Two quantum pieces split from two sources could not be merge into to one.
+    board = set_board(["a1", "b1"])
+    world = board.board
+    SplitJump()(world["a1"], world["a2"], world["a3"])
+    SplitJump()(world["b1"], world["b2"], world["b3"])
+    MergeJump()(world["a2"], world["b2"], world["c2"])
+    # According to matrix calculations, the ending coefficient of
+    # [a3, b3]: 1/2
+    # [a3, c2]: i/2/sqrt(2)
+    # [a3, b2]: -1/2/sqrt(2)
+    # [b3, c2]: i/2/sqrt(2)
+    # [b2, b3]: 1/2/sqrt(2)
+    # [b2, c2]: 1/2
+    assert_sample_distribution(
+        board,
+        {
+            locations_to_bitboard(["a3", "b3"]): 1.0 / 4,
+            locations_to_bitboard(["a3", "c2"]): 1.0 / 8,
+            locations_to_bitboard(["a3", "b2"]): 1.0 / 8,
+            locations_to_bitboard(["b3", "c2"]): 1.0 / 8,
+            locations_to_bitboard(["b2", "b3"]): 1.0 / 8,
+            locations_to_bitboard(["b2", "c2"]): 1.0 / 4,
+        },
+    )
+
+
+def test_merge_jump_imperfect_merge_scenario_3():
+    # Imperfect merge scenario 3
+    # This is a simplied version of the scenario above, where we unhook a3 and b3.
```

Comment:
Consider changing the line comments here and above to docstrings with triple quotes.  I think they will show up in IDEs better.

Reply:
Thanks Doug! It's updated.